src/config/config_type.rs

```diff
@@ -49,7 +74,63 @@ impl ConfigType<'_> {
                         err.to_string()
                     ))
                 })
-                .and_then(|config| T::try_from(config).map_err(Error::ConvertProtoConfig)),
+                .and_then(|config| Static::try_from(config).map_err(Error::ConvertProtoConfig))
+                .and_then(|config| Self::get_json_config(filter_name, config)),
         }
     }
+
+    // Returns an equivalent json value for the passed in config.
+    fn get_json_config<T>(filter_name: &str, config: T) -> Result<(serde_json::Value, T), Error>
+    where
+        T: serde::Serialize + for<'de> serde::Deserialize<'de>,
+    {
+        serde_json::to_string(&config)
+            .map_err(|err| {
+                Error::DeserializeFailed(format!(
+                    "filter `{}`: failed to serialize config to json: {}",
+                    filter_name,
+                    err.to_string()
+                ))
+            })
+            .and_then(|config_json| {
+                serde_json::from_str::<serde_json::Value>(config_json.as_str()).map_err(|err| {
+                    Error::DeserializeFailed(format!(
+                        "filter `{}`: failed to deserialize into json: {}",
+                        filter_name,
+                        err.to_string()
+                    ))
+                })
+            })
+            .map(|config_json| (config_json, config))
```

Comment:
If you take it by reference here, you don't need to pass it through the function and back to the caller, and we can also remove level of serialisation.

```suggestion
    fn get_json_config<T>(filter_name: &str, config: &T) -> Result<serde_json::Value, Error>
    where
        T: serde::Serialize + for<'de> serde::Deserialize<'de>,
    {
        serde_json::to_value(config.clone())
            .map_err(|err| {
                Error::DeserializeFailed(format!(
                    "filter `{}`: failed to serialize config to json: {}",
                    filter_name,
                    err.to_string()
                ))
            })
```